File: Two Equal Sum Subarrays.java
import java.util.Arrays;

class Solution {
    public boolean canSplit(int arr[]) {
        int sum = Arrays.stream(arr).sum();
        // code here 
        if (sum % 2 == 1) return false;
        
        int target = sum / 2;
        int runner = 0;
        
        for (int i = 0; i < arr.length - 1; i++) { 
            runner += arr[i];
            if (runner == target) {
                return true;
            }
        }
        return false;
    }
}

File: Two Equal Sum Subarrays.py
class Solution:
    def canSplit(self, arr):
        #code here
        total = sum(arr)
        
        if total % 2 == 1:
            return False
        
        target = total // 2
        runner = 0
    
        for i in range(len(arr) - 1):
            runner += arr[i]
            if runner == target:
                return True
                
        return False
